agent/agent-bootstrap/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/TemporaryMetricsView.java

```diff
@@ -0,0 +1,147 @@
+    Set<AttributeKey> view = new HashSet<>();
+    // START APPLICATION INSIGHTS MODIFICATIONS
+    // view.add(SemanticAttributes.HTTP_METHOD);
+    view.add(SemanticAttributes.HTTP_STATUS_CODE); // Optional
+    // view.add(SemanticAttributes.HTTP_FLAVOR); // Optional
+    // END APPLICATION INSIGHTS MODIFICATIONS
+    return view;
+  }
+
+  private static Set<AttributeKey> buildDurationClientView() {
+    // We pull identifying attributes according to:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attribute-alternatives
+    // We only pull net.peer.name and net.peer.port because http.url has too high cardinality
+    Set<AttributeKey> view = new HashSet<>(durationAlwaysInclude);
+    view.add(SemanticAttributes.NET_PEER_NAME);
+    view.add(SemanticAttributes.NET_PEER_PORT);
+    return view;
+  }
+
+  private static Set<AttributeKey> buildDurationServerView() {
+    // We pull identifying attributes according to:
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attribute-alternatives
+    // With the following caveat:
+    // - we always rely on http.route + http.host in this repository.
+    // - we prefer http.route (which is scrubbed) over http.target (which is not scrubbed).
+    Set<AttributeKey> view = new HashSet<>(durationAlwaysInclude);
+    // START APPLICATION INSIGHTS MODIFICATIONS
+    // view.add(SemanticAttributes.HTTP_SCHEME);
+    // view.add(SemanticAttributes.HTTP_HOST);
+    // view.add(SemanticAttributes.HTTP_ROUTE);
+    // END APPLICATION INSIGHTS MODIFICATIONS
+    return view;
+  }
+
+  private static Set<AttributeKey> buildActiveRequestsView() {
+    // the list of included metrics is from
+    // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes
+    Set<AttributeKey> view = new HashSet<>();
+    view.add(SemanticAttributes.HTTP_METHOD);
+    view.add(SemanticAttributes.HTTP_HOST);
+    view.add(SemanticAttributes.HTTP_SCHEME);
+    view.add(SemanticAttributes.HTTP_FLAVOR);
+    view.add(SemanticAttributes.HTTP_SERVER_NAME);
+    return view;
+  }
+
+  // START APPLICATION INSIGHTS MODIFICATIONS
+
+  static Attributes applyClientDurationAndSizeView(
+      Context context, Attributes startAttributes, Attributes endAttributes) {
+    AttributesBuilder filtered = Attributes.builder();
+    applyView(filtered, startAttributes, durationClientView);
+    applyView(filtered, endAttributes, durationClientView);
+
+    PreAggregatedStandardMetrics.applyHttpClientView(
+        filtered, context, startAttributes, endAttributes);
+
+    return filtered.build();
+  }
+
+  // had to add "2" to this method name because sometimes compilation picks up the upstream version
+  static Attributes applyServerDurationAndSizeView2(
```

Comment:
why it would pick up the upstream version, which is excluded?

Reply:
the problem was happening to me sometimes during compilation, because we don't exclude it until shadowJar, which is after compilation